benchmarks/paket.references

```diff
@@ -1,4 +1,3 @@
-FSharp.Core
```

Comment:
Are you sure we want to remove this entry?
Don't we want to have the explicit `FSharp.Core` reference?

Reply:
I can add it back after work. I made the assumption paket knew better than me. Silly question but why would we need to keep it if paket is saying it is 'unused'? Is this a paket bug you think?

Reply:
Hmm, that sounds a bit weird to me. Isn't F# Core always used? The compiler needs it to compile your code right?

Reply:
it's marked as a transitive dependency. I know project files have some special rules around FSharp.Core (implicit vs explicit). Probably just need to look at `project.assets.json` and see if it's using the transitive defined in paket.lock or the sdk version

Reply:
@1eyewonder could you log an issue on paket, with a bit of info you have at hand to reproduce it (the more the merrier), I think any paket.references to F# project that references FSharp.Core should have this line in it, and we can move this discussion to paket repository to gather more insight on what should occur.

Reply:
Yep, I can get on that later this evening. As for this PR, we are in agreeance to add FSharp.Core back in the project.references correct?

Reply:
Left is paket.lock, right is project.assets.json for src/FsAutoComplete
![image](https://github.com/fsharp/FsAutoComplete/assets/38481538/ca44d8c1-a93e-4359-af8b-f8c77f30f8e2)

Left is paket.lock, right is project.assets.json for benchmarks
![image](https://github.com/fsharp/FsAutoComplete/assets/38481538/fa13798d-8a5b-4034-a843-9d76e4fe1ab2)

I think we should be good to proceed